src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/AsyncDisposable.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using IAsyncDisposable = System.IAsyncDisposable;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    internal sealed class AsyncDisposable : IAsyncDisposable
```

Comment:
This seems like a good utility that should be outside of the ProjectSystem scope, since it isn't specific to it.

Reply:
This would be available to pretty much all code in the project system.

I'm not sure there's enough value here to make it more generally available. You might also expect it to be implemented in terms of `Task` rather than `ValueTask`.